### PR TITLE
posix_memalign for O_DIRECT

### PR DESCRIPTION
**Pull request by riverszhang89**
Ensure that we're reading into a buffer aligned on a 512-byte boundary, for O_DIRECT analyze.

### CODE REVIEW

**Review by mponomar — APPROVED** [review]

**Comment by akshatsikarwar** [comment]
While you are at it, can you also fix this:
```
summarize.c", line 349: error: undefined symbol: O_DIRECT
```

Need to call `directio()` on SunOS instead.

File: bdb/summarize.c [hunk 1]
```diff
@@ -358,8 +363,13 @@ int bdb_summarize_table(bdb_state_type *bdb_state, int ixnum, int comp_pct,
         goto done;
     }
 
-    rc = read(fd, metabuf, sizeof(metabuf));
-    if (rc != sizeof(metabuf)) {
+    /* O_DIRECT: ensure the buffer is aligned on a 512-byte boundary. */
+    rc = posix_memalign(&metabuf, 512, METABUFLEN);
```
Comment (markhannum) [inline bdb/summarize.c:367]: is metabuf free'd later?

Comment (riverszhang89) [reply bdb/summarize.c:367]: Yes, it needs to be freed later.

**Comment by riverszhang89** [comment]
Closing this PR now as Mike has #4236 a much smaller change for the same issue.

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Error &#x26A0;.<br>Regression testing: <b>376/520 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>logfill_logput_window_generated<br>truncatesc_ufid_on_generated<br>queuedb_rollover_noroll1_generated<br>yast_stat4scan_generated<br>yast<br>prepare<br>analyze_recover_deadlock<br>autoanalyze<br>analyze_partial_index_off_generated<br>analyze</p>

**Pull request closed without merging**